UISearchBar.cs

```diff
@@ -181,10 +183,20 @@ protected override void DrawSelf(SpriteBatch spriteBatch)
                 color *= 0.75f;
             }
             spriteBatch.DrawString(font, drawText, new Vector2(dim.X + padding, dim.Y + padding), color, 0f, Vector2.Zero, scale, SpriteEffects.None, 0f);
-            if (!isEmpty && hasFocus && cursorTimer < 30)
+            if (!isEmpty && hasFocus)
             {
+                Main.instance.DrawWindowsIMEPanel(new Vector2(24f, 316f), 0f); // IME panel drawing. Necessary for inputting Chinese or some languages.
```

Comment:
Why specifically `new Vector2(24f, 316f)`?

Reply:
The IME panel will be drawn on the top of the main storage panel. Try to avoid being blocked by other panels.

Reply:
Are you sure the position is correct for other resolutions?

Reply:
In `CraftingGUI.Initialize()` and `StorageGUI.Initialize()`:
```
panelTop = Main.instance.invBottom + 60;
panelLeft = 20f;
```
The value of `Main.instance.invBottom` is never changed in vanllia Terraria.
In `Main.Initialize()`:
```
invBottom = 258
```
However, in case other mods change it, I will change it to `new Vector2(20f, Main.instance.invBottom + 60)`